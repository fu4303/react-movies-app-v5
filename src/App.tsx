import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from 'themes';
import { Navbar, Footer } from 'components';
import { HomePage, MovieList } from 'screens';
import { GlobalStyle } from 'components/global-styles';

import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';

const App = () => (
  <ThemeProvider theme={theme}>
    <GlobalStyle fontColor={theme.light} bodyColor={theme.darkDeepest} />
    <Router>
      <Navbar title="MOVIES DEV" />
      <Switch>
        <Route exact path="/">
          <HomePage />
        </Route>
        <Route path="/search">
          <MovieList pageTitle="Searching" />
        </Route>
        <Route path="/favorites">
          <MovieList pageTitle="My Favorites" />
        </Route>
        <Route path="/watchlist">
          <MovieList pageTitle="My Watch List" />
        </Route>
      </Switch>
      <Footer />
    </Router>
  </ThemeProvider>
);

export default App;
