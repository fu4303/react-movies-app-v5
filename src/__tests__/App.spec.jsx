import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from '../App';

afterEach(cleanup);

describe('App tests', () => {
  it('Should match the snapshot for "App" component', () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
});
