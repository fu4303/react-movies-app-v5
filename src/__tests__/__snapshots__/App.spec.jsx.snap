// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Should match the snapshot for "App" component 1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw iAA-dCc"
  >
    <nav
      class="sc-AxhCb jvvfzj"
    >
      <a
        aria-current="page"
        class="sc-AxjAm eNfybe active"
        href="/"
      >
        <div
          class="sc-AxgMl cjEjyT"
        >
          <img
            alt="Movie icon"
            class="sc-Axmtr koDVaQ"
            src="movie_icon.svg"
          />
          <h1
            class="sc-AxheI kouLZm"
          >
            MOVIES DEV
          </h1>
        </div>
      </a>
      <ul
        class="sc-AxhUy cevNGS"
      >
        <li
          class="sc-AxmLO oZQsq"
        >
          <a
            class="sc-AxjAm sc-AxirZ bWMUTY"
            href="/search"
          >
            Search
          </a>
        </li>
        <li
          class="sc-AxmLO oZQsq"
        >
          <a
            class="sc-AxjAm sc-AxirZ bWMUTY"
            href="/watchlist"
          >
            Watch Later
          </a>
        </li>
        <li
          class="sc-AxmLO oZQsq"
        >
          <a
            class="sc-AxjAm sc-AxirZ bWMUTY"
            href="/favorites"
          >
            Favorites
          </a>
        </li>
      </ul>
    </nav>
  </div>
  loading...
  <h2
    class="sc-fzoXzr kCmOKr"
  >
     More on Popular 
  </h2>
  <section
    class="sc-fzoyAV gMhNNv"
  />
  <footer
    class="sc-fzoXWK ktEZmg"
  >
    <hr />
    Â©2020, All rights reserved.
  </footer>
</DocumentFragment>
`;
